packages/realm/src/index.ts

```diff
@@ -18,44 +18,110 @@
 
 export {
   App,
+  AppChangeCallback,
+  AppConfiguration,
+  BaseConfiguration,
+  BaseObjectSchema,
+  BaseSubscriptionSet,
+  BaseSyncConfiguration,
   BSON,
+  CanonicalObjectSchemaProperty,
+  CanonicalPropertiesTypes,
+  ClientResetAfterCallback,
+  ClientResetBeforeCallback,
+  ClientResetConfig,
+  ClientResetDiscardUnsyncedChangesConfiguration,
+  ClientResetFallbackCallback,
+  ClientResetManualConfiguration,
   ClientResetMode,
+  ClientResetRecoverOrDiscardUnsyncedChangesConfiguration,
+  ClientResetRecoverUnsyncedChangesConfiguration,
+  /**
+   * This type got renamed to {@link ClientResetRecoverUnsyncedChangesConfiguration}
+   * @deprecated Please use named imports
+   */
+  ClientResetRecoverUnsyncedChangesConfiguration as ClientResetRecoveryConfiguration,
   Collection,
+  CollectionChangeCallback,
+  CollectionChangeSet,
+  Configuration,
+  ConfigurationWithoutSync,
+  ConfigurationWithSync,
+  ConnectionNotificationCallback,
+  ConnectionState,
   Credentials,
+  DefaultFunctionsFactory,
+  DefaultUserProfileData,
   Dictionary,
+  DictionaryChangeCallback,
+  DictionaryChangeSet,
+  ErrorCallback,
+  flags,
+  FlexibleSyncConfiguration,
   List,
+  LocalAppConfiguration,
+  LogLevel,
+  MigrationCallback,
   MutableSubscriptionSet,
+  NumericLogLevel,
+  /**
+   * This type got renamed to {@link RealmObjectConstructor}
+   * @deprecated Please use named imports
+   */
+  RealmObjectConstructor as ObjectClass,
+  ObjectChangeCallback,
+  ObjectChangeSet,
+  ObjectSchema,
+  OpenRealmBehaviorConfiguration,
+  OpenRealmBehaviorType,
+  OpenRealmTimeOutBehavior,
   OrderedCollection,
+  PartitionSyncConfiguration,
+  PartitionValue,
+  PrimaryKey,
+  ProgressDirection,
+  ProgressMode,
+  ProgressNotificationCallback,
   ProgressRealmPromise,
+  PropertiesTypes,
+  PropertySchema,
+  ObjectSchemaProperty,
+  PropertySchemaShorthand,
+  ProviderType,
   Realm,
+  RealmFunction,
   RealmObject as Object,
+  RealmObjectConstructor,
   RealmSet as Set,
   Results,
+  SessionState,
   SessionStopPolicy,
+  SortDescriptor,
   Subscription,
+  SubscriptionOptions,
   SubscriptionSet,
   SubscriptionsState,
+  SyncConfiguration,
+  SyncError,
+  SyncSession,
+  /**
+   * @deprecated Got renamed to {@link SyncSession} and please use named imports
+   */
+  SyncSession as Session,
+  Types,
   UpdateMode,
   User,
-  Types,
-  flags,
+  UserChangeCallback,
+  UserState,
+  EmailPasswordAuth,
+  ApiKey,
+  ApiKeyAuth,
 } from "./internal";
 
-export type {
-  CollectionChangeCallback,
-  CollectionChangeSet,
-  Configuration,
-  FlexibleSyncConfiguration,
-  ObjectChangeCallback,
-  ObjectChangeSet,
-  ObjectSchema,
-  PropertySchema,
-  PartitionSyncConfiguration,
-  RealmEventName,
-  RealmListenerCallback,
-  SubscriptionOptions,
-} from "./internal";
```

Comment:
@elle-j I remember you mentioned that you had errors when exporting these without the `type` keyword. Do you remember when that error occurred? This builds for me without issues, so I wonder if it was somewhere when running the tests?

Reply:
I've made a few changes and verified that switching to tsx as the TS runtime makes the SDK unit tests run again: https://github.com/realm/realm-js/pull/5378